numismatic/feeds/base.py

```diff
@@ -351,7 +323,17 @@ def __handle_packet(packet, subscription):
             raise
         if not msg:
             return
-        for handler in subscription.handlers:
-            result = handler(msg, subscription)
-            if result is STOP_HANDLERS:
-                break
+        for subscription in self.subscriptions:
+            for handler in subscription.handlers:
+                result = handler(msg, subscription)
+                if result is STOP_HANDLERS:
+                    # FIXME: should this raw_stream now rather sit on the
+                    # WebsocketClient instead of the Subscription?
+                    subscription.raw_stream.emit(packet)
```

Comment:
The raw `packet`s now still go to the subscriptions but I think they should really be associated with the websocket_client. So we need to move the `raw_stream` from the subscription to the websocket_client. I think that's probably best handled in a separate PR though.

Reply:
Yes I agree. In general, it seems like Subscription is trying to do too much.

Reply:
Beyond that, I see the main purpose of capturing the raw packets for debugging. As such they should be captured as close to the source as possible. By the time they hit a subscription they've already been filtered and assigned to a subscription. If that logic is faulty then having the raw packet end up in the wrong place isn't really going to help with the debugging. Also, you might lose the sequencing implicit in the packet arrival this way which is another downside.

I'll do this refactoring in separate PR.